design-proposals/dynamic-primary-pod-nic-name.md

```diff
@@ -0,0 +1,466 @@
+Support Dynamic Primary Pod NIC Name
+=
+
+# Summary
+
+This proposal proposes a mechanism to dynamically determine the primary pod interface for KubeVirt virtual machines (VMs).
+The goal is to move away from the hard coded `eth0` interface and allow flexible, environment-driven interface selection.
+
+# Overview
+
+## Motivation
+
+In KubeVirt, each VM runs as a Kubernetes pod, and these pods have network interfaces attached to different networks.
+The default primary pod interface is often named `eth0`.
+Currently, KubeVirt attaches the VM’s interface destined to the pod network to the pod’s `eth0` interface.
+However, users may require custom network setups where the pod’s primary interface is not named `eth0`.
+
+## Goals
+
+- Allow the primary pod interface name to be dynamically determined based on the environment.
+- Ensure backward compatibility, so existing VMs with `eth0` as the primary interface will not break.
+
+## Non Goals
+
+- Migration of an existing VM whose pod primary interface is `eth0` to a pod whose primary interface has a custom name and vise versa. 
+
+## Definition of Users
+
+- VM owner
+
+## User Stories
+
+As a VM owner:
+- I want my existing VMs using `eth0` as their primary pod interface to continue working.
```

Comment:
As far as I can tell from the perspective of a user they don't even know that `eth0` was used before? IMO the third item suffices to describe it.

Reply:
Done.